Set/Models/GameModel.swift

```diff
@@ -94,25 +98,47 @@ struct GameModel {
         }
         
         if checkASet(for: selectedCards) {
+            let currentSuccesfullMatchDate: Date = .now
+            let currentTurnDuration: TimeInterval = currentTurnDuration(for: currentSuccesfullMatchDate)
+            
             mark(selectedCards, as: .isMatchedSuccessfully)
+            
+            // I can move this logic to a method, but I don't know how to call it
+            if player.id == firstPlayer.id {
+                firstPlayer.increaseScore(by: calculateBonus(for: currentTurnDuration))
+            } else if player.id == secondPlayer.id {
+                secondPlayer.increaseScore(by: calculateBonus(for: currentTurnDuration))
+            }
+            // ------------------------------------------------------------------
+            
+            previousSuccesfullMatchDate = currentSuccesfullMatchDate
+            previousTurnDuration = currentTurnDuration
+            
             return .successfulMatch
         } else {
             mark(selectedCards, as: .isMatchedUnsuccessfully)
+            
+            // I can move this logic to a method, but I don't know how to call it
```

Comment:
You can do it same way as it was before, in method `getBonus` but name it like `increaseScore(playerID: UUID, value: Int)`

Reply:
done